### PR TITLE
x11-libs/libxshmfence: unbreak build on freebsd 11.0+

### PR DESCRIPTION
**Pull request by stha09**
Package-Manager: Portage-2.3.19, Repoman-2.3.6

### CODE REVIEW

**Comment by stha09** [comment]
Please ping upstream to merge patch from https://patchwork.freedesktop.org/patch/165927/ (I dont have account there)

File: x11-libs/libxshmfence/libxshmfence-1.2-r1.ebuild [hunk 1]
```diff
@@ -0,0 +1,19 @@
+# Copyright 1999-2018 Gentoo Foundation
+# Distributed under the terms of the GNU General Public License v2
+
+EAPI=5
```
Comment (doughdemon) [inline x11-libs/libxshmfence/libxshmfence-1.2-r1.ebuild]: Please convert to EAPI 6.

File: x11-libs/libxshmfence/libxshmfence-1.2-r1.ebuild [hunk 2]
```diff
@@ -0,0 +1,19 @@
...
+
+KEYWORDS="~alpha ~amd64 ~arm ~arm64 ~hppa ~ia64 ~mips ~ppc ~ppc64 ~s390 ~sh ~sparc ~x86 ~amd64-fbsd ~x86-fbsd ~amd64-linux ~arm-linux ~x86-linux ~sparc-solaris ~x64-solaris ~x86-solaris"
+IUSE=""
+
+RDEPEND=""
+DEPEND=">=x11-proto/xproto-7.0.24:=[${MULTILIB_USEDEP}]
+	${RDEPEND}"
+
+src_prepare() {
+	epatch "${FILESDIR}"/${P}-freebsd-futex.patch
```
Comment (doughdemon) [inline x11-libs/libxshmfence/libxshmfence-1.2-r1.ebuild]: Please use the PATCHES array.

File: x11-libs/libxshmfence/libxshmfence-1.2-r1.ebuild [hunk 3]
```diff
@@ -0,0 +1,19 @@
...
+EAPI=5
+XORG_MULTILIB=yes
+inherit eutils xorg-2
+
+DESCRIPTION="Shared memory fences using futexes"
+
+KEYWORDS="~alpha ~amd64 ~arm ~arm64 ~hppa ~ia64 ~mips ~ppc ~ppc64 ~s390 ~sh ~sparc ~x86 ~amd64-fbsd ~x86-fbsd ~amd64-linux ~arm-linux ~x86-linux ~sparc-solaris ~x64-solaris ~x86-solaris"
+IUSE=""
+
+RDEPEND=""
```
Comment (doughdemon) [inline x11-libs/libxshmfence/libxshmfence-1.2-r1.ebuild]: This is not needed.

File: x11-libs/libxshmfence/libxshmfence-1.2-r1.ebuild (hunk 1 above)
Comment (stha09) [reply x11-libs/libxshmfence/libxshmfence-1.2-r1.ebuild]: xorg-2.eclass has no support for EAPI 6

**Comment by doughdemon** [comment]
> xorg-2.eclass has no support for EAPI 6

I see, sorry about my wrong suggestion.

Still, if epatch is the only thing you use from eutils, you could
convert to epatch.eclass.

**Comment by stha09** [comment]
2nd round

**Comment by mattst88** [comment]
Please send the patch to xorg-devel@lists.x.org. Once upstream, I can make a new libxshmfence release and we won't have to carry patches in Gentoo.

**Comment by stha09** [comment]
I really don't have time for this, I'm just helping out.

**Comment by mattst88** [comment]
That's the spirit!

**Comment by mattst88** [comment]
Patch from https://bugs.gentoo.org/648548 is now upstream, and I'll make a tarball release soon.

**Pull request closed without merging**